schema/elements.go

```diff
@@ -180,15 +184,31 @@ type Map struct {
 type Untyped struct {
 	// ElementRelationship states the relationship between the items, if
 	// container-typed data happens to be present here.
+	// * `lookup` calls a lookup function to figure out the
+	//    schema based on the data. lookupFunc must be set.
 	// * `atomic` implies that all elements depend on each other, and this
 	//   is effectively a scalar / leaf field; it doesn't make sense for
 	//   separate actors to set the elements.
 	// TODO: support "guess" (guesses at associative list keys)
-	// TODO: support "lookup" (calls a lookup function to figure out the
-	//       schema based on the data)
 	// The default behavior for untyped data is `atomic`; it's permitted to
 	// leave this unset to get the default behavior.
 	ElementRelationship ElementRelationship `yaml:"elementRelationship,omitempty"`
+
+	// lookupFunc should only be set if elementRelationship is `lookup`.
+	lookupFunc LookupFunc
```

Comment:
This is not serialized and/or persisted, which means that the behavior is sort of unpredictable?